### PR TITLE
Messaging flow 57b

### PR DESCRIPTION
**Pull request by cch5ng**
not sure if I should get a review now and do the remaining tasks in another ticket, but I think my list of changes keeps growing here

**done:**
* majority of UI for the chat view is done; some styling needs to be fixed and timestamps need to be made dynamic
* right now for a hard-coded conversation (existing in the db), I can get a 2way chat to display on the screen
* right now for a hard-coded conversation (existing in db), new chat messages will be saved to the conversation doc
* update conversation model to reference user emails (for contacts I know email is available but id is not currently)
* refactored auth context/login to pull most of the user data and save it to localstorage
* made small route workarounds so on login/logout, the routes display appropriately
* added a control to test logout (it is kind of a placeholder; the horizontal ellipsis upper right corner)
   * Rajat wrote the signout() function but I was working on UI for the control so I just helped to hook it up 

**not done:**
* all of the new chat setup
* functionality to create a new conversation doc (click on contact)
* flow to redirect to a new conversation specific route (route is defined)
* definition of socket room for private chat
* final integration testing for saving messages (with dynamic conversation id)

### CODE REVIEW

File: client/src/components/Chat/ChatHeader.js [hunk 1]
```diff
@@ -1,34 +1,129 @@
...
+    backgroundColor: theme.palette.common.white,
+  },
+  checked: {},
+}))(Switch);
 
 const ChatHeader = props => {
-  const {selectedContact} = props;
+  const {selectedContacts} = props;
+  const classes = useStyles();
+  const [state, setState] = React.useState({
```
Comment (bonnieli) [inline client/src/components/Chat/ChatHeader.js:73]:
```suggestion
  const [state, setState] = useState({
```

File: client/src/components/Chat/ChatHeader.js [hunk 2]
```diff
@@ -1,34 +1,129 @@
...
+  const {user, logout} = useAuth();
+  let initial;
+  let email;
+  if (selectedContacts.length === 1) {
+    initial = selectedContacts[0].email[0].toUpperCase();
+    email = selectedContacts[0].email;
+  };
+
+  const handleChange = (event) => {
+    setState({ ...state, [event.target.name]: event.target.checked });
```
Comment (bonnieli) [inline client/src/components/Chat/ChatHeader.js:85]:
```suggestion
  const {name, checked} = event.target;
    setState({ ...state, [name]: checked });
```

**Review by bonnieli — COMMENTED** [review]
this lgtm, i tested this locally and looks like it's set up correctly!

File: client/src/components/Chat/ChatHeader.js (hunk 1 above)
Comment (cch5ng) [reply client/src/components/Chat/ChatHeader.js:73]: hi @bonnieli do you mind if I make these changes in my current working branch, #65 ?

Comment (cch5ng) [reply client/src/components/Chat/ChatHeader.js:73]: @bonnieli I made both changes in my working branch, #65 and pushed to the open pr. I still have some work to do in that pr.

**Pull request merged**